salt/metalk8s/kubernetes/apiserver/installed.sls

```diff
@@ -21,6 +21,12 @@ Set up default basic auth htpasswd:
     - replace: False
 
 {%- set host = grains['metalk8s']['control_plane_ip'] %}
+{%- set etcd_servers = [] %}
+{%- for member in pillar.metalk8s.etcd.members | default([]) %}
+{%-   do etcd_servers.extend(member['client_urls']) %}
+{%- endfor %}
+{%- set etcd_servers =
+      (["https://" ~ host ~ ":2379"] + etcd_servers | sort) | unique %}
```

Comment:
with the sort doesn't that mean that sometimes the first etcd server kubeapiserver will hit maybe not the local one ? also  did you check when killing the first etcd in the list that kubeapiserver fall over to the next ? I can see some bugs 
here https://github.com/kubernetes/kubernetes/issues/83028 and here https://github.com/kubernetes/kubernetes/issues/72102#issuecomment-460303835

Reply:
sorry used my personal account again :p

Reply:
> with the sort doesn't that mean that sometimes the first etcd server kubeapiserver will hit maybe not the local one ?

The `etcd_servers` variable will always have the local member in first position, so it shouldn't change this behaviour. However, now that I think of it, we should *only* add this _local member_ *if* it has the `etcd` role. Something like:

```sls
{%- if 'etcd' in pillar.metalk8s.nodes[grains.id].roles %}
{%-   do etcd_servers.insert(0, "https://" ~ host ~ ":2379") %}
{%- endif %}
```

> also did you check when killing the first etcd in the list that kubeapiserver fall over to the next? I can see some bugs here kubernetes/kubernetes#83028 and here kubernetes/kubernetes#72102 (comment)

Indeed, @alexandre-allard-scality is aware of this issue. It is supposed to be fixed in K8s 1.16, from the issues you linked.
If we were to have a failing etcd member, the apiserver deployed on that node will not work either, and this is current behaviour (as we only list the assumed local member). In 2.5 and above, we _should_ have a working fallback to other `etcd` members.
I think it would make sense to add a BDD test for this, and mark it as expected fail if the version is lower than when the fix was introduced. Do you think you can squeeze something like this in this PR @alexandre-allard-scality?

Reply:
oh yes sort is only applied to the remote list mb !

Reply:
I'd add some parenthesis to make the desired outcome more clear. Precedence between `+` and `|` is not necessarily... 'obvious'.

What we *want* (I guess?) is
```
{%- set etcd_servers =
      (["https://" ~ host ~ ":2379"] + (etcd_servers | sort)) | unique %}
```
This may be the case without the extra parenthesis (and operator precedence could be inferred from the use of parenthesis to bind `| unique` to the combined list), but I prefer to be explicit.

Though not a blocker for this PR as far as I'm concerned :man_shrugging:

Reply:
Applied comment of @gdemonet, we don't need extra parenthesis with this change.